src/ray/gcs/gcs_server/gcs_actor_manager.cc

```diff
@@ -205,17 +205,34 @@ const ray::rpc::ActorDeathCause GcsActorManager::GenNodeDiedCause(
     const std::string ip_address,
     std::shared_ptr<rpc::GcsNodeInfo> node) {
   ray::rpc::ActorDeathCause death_cause;
+
   auto actor_died_error_ctx = death_cause.mutable_actor_died_error_context();
   AddActorInfo(actor, actor_died_error_ctx);
-  actor_died_error_ctx->set_error_message(
-      absl::StrCat("The actor is dead because its node has died. Node Id: ",
-                   NodeID::FromBinary(node->node_id()).Hex()));
-
-  // TODO(vitsai): Publish this information as well
-  if (auto death_info = node->death_info();
-      death_info.reason() == rpc::NodeDeathInfo::AUTOSCALER_DRAIN_PREEMPTED) {
-    actor_died_error_ctx->set_preempted(true);
+  auto node_death_info = actor_died_error_ctx->mutable_node_death_info();
+  node_death_info->CopyFrom(node->death_info());
+
+  std::string error_message = "The actor died because its node has died. Node Id: " +
```

Comment:
should we use ostringstream to construct the error message?

Reply:
Either should be OK? Updated with ostringstream.

Reply:
yea, I guess it doesn't matter that much if we only append string